tools/sgx-lkl-docker

```diff
@@ -44,22 +44,46 @@ function build() {
 FROM ubuntu:18.04
 RUN apt-get update && apt-get install -y ca-certificates
 
-FROM scratch
 # ca-certificates.crt is read by libcurl which is used by the Azure DCAP Client.
-COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
-COPY . /
 
-# Note: WORKDIR also creates the directory.
-WORKDIR /tmp
-ENV TMPDIR=/tmp
+RUN groupadd -g 999 sgxlkl && \
+    useradd -g 999 -u 999 sgxlkl
+
+RUN mkdir /prep && \
+    cd /prep && \
+    mkdir etc && \
+    echo "sgxlkl:x:999:999::/tmp:/usr/sbin/nologin" > etc/passwd && \
+    echo "sgxlkl:x:999:" > etc/group && \
+    mkdir -p home/sgxlkl && \
+    chown sgxlkl:sgxlkl home/sgxlkl && \
```

Comment:
I wonder how robust that is. Isn't it better to invoke `useradd` with the correct parameters?

I actually don't understand why you do this as part of a multi-stage build and not directly in the target container.

Reply:
Which part do you think is not robust here? The contents of `/etc/passwd` and `/etc/group` are standard and Docker itself parses these files if you use the name form of `-u user`.
The reason I'm creating the files/folders manually is that I want them to be minimal and independent of the first stage base image (Ubuntu).

Reply:
I'm worried when files like this are edited directly. For example, what about if there is a clash with uid 999?

Reply:
Normal UIDs start at 1000. And in a standard Ubuntu base image there is only uid 0 anyway.